File: backend/src/main/java/com/NBE3_4_2_Team4/global/security/jwt/JwtObjectMapper.java
//package com.NBE3_4_2_Team4.global.security.jwt;
//
//import com.NBE3_4_2_Team4.domain.member.member.entity.Member;
//import com.NBE3_4_2_Team4.standard.constants.AuthConstants;
//import org.springframework.stereotype.Component;
//
//import java.util.Map;
//
//@Component
//public class JwtObjectMapper {
//    public Member getMemberByJwtClaims(Map<String, Object> claims) {
//        Integer id = (Integer) claims.get(AuthConstants.ID);
//        String username = (String) claims.get(AuthConstants.USERNAME);
//        String nickname = (String) claims.get(AuthConstants.NICKNAME);
//        String roleName = (String) claims.get(AuthConstants.ROLE);
//        String OAuth2ProviderName = (String) claims.get(AuthConstants.OAUTH2_PROVIDER);
//        String emailAddress = (String) claims.get(AuthConstants.EMAIL_ADDRESS);
//        Boolean emailVerified = (Boolean) claims.get(AuthConstants.EMAIL_VERIFIED);
//
//        if (id == null
//                || isNullOrBlank(username)
//                || isNullOrBlank(nickname)
//                || isNullOrBlank(roleName)
//                || isNullOrBlank(OAuth2ProviderName)
//                || isNullOrBlank(emailAddress)
//                || emailVerified == null) {
//            throw new RuntimeException("Invalid claims");
//        }
//
//        return new Member(Long.valueOf(id), username, nickname, roleName, OAuth2ProviderName, emailAddress, emailVerified);
//    }
//
//    private boolean isNullOrBlank(String string) {
//        return string == null || string.isBlank();
//    }
//}

File: backend/src/main/java/com/NBE3_4_2_Team4/global/security/jwt/JwtObjectMapper.kt
package com.NBE3_4_2_Team4.global.security.jwt

import com.NBE3_4_2_Team4.domain.member.member.entity.Member
import com.NBE3_4_2_Team4.standard.constants.AuthConstants
import org.springframework.stereotype.Component

@Component
class JwtObjectMapper {
    fun getMemberByJwtClaims(claims: Map<String?, Any?>): Member {
        val id = claims[AuthConstants.ID] as Int?
        val username = claims[AuthConstants.USERNAME] as String?
        val nickname = claims[AuthConstants.NICKNAME] as String?
        val roleName = claims[AuthConstants.ROLE] as String?
        val oAuth2ProviderName = claims[AuthConstants.OAUTH2_PROVIDER] as String?
        val emailAddress = claims[AuthConstants.EMAIL_ADDRESS] as String?
        val emailVerified = claims[AuthConstants.EMAIL_VERIFIED] as Boolean?

        if ((id == null || username.isNullOrBlank()
                    || nickname.isNullOrBlank()
                    || roleName.isNullOrBlank()
                    || oAuth2ProviderName.isNullOrBlank()
                    || emailAddress.isNullOrBlank()
                    || emailVerified == null)) {
            throw RuntimeException("Invalid claims")
        }

        return Member.from(
            id = id.toLong(),
            username = username,
            nickname = nickname,
            roleName = roleName,
            oAuth2ProviderName = oAuth2ProviderName,
            emailAddress = emailAddress,
            emailVerified = emailVerified
        )
    }
}